plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/inject/LanguageServerGinModule.java

```diff
@@ -45,14 +45,10 @@ protected void configure() {
         install(new GinFactoryModuleBuilder().build(LanguageServerQuickAssistProcessorFactory.class));
         install(new GinFactoryModuleBuilder().build(LanguageServerReconcileStrategyFactory.class));
         install(new GinFactoryModuleBuilder().build(LanguageServerSignatureHelpFactory.class));
-        bind(LanguageServerRegistry.class);
 
         GinMapBinder<String, WsAgentComponent> wsAgentComponentsBinder =
                 GinMapBinder.newMapBinder(binder(), String.class, WsAgentComponent.class);
         wsAgentComponentsBinder.addBinding("Load Language Server file types.").to(LanguageServerFileTypeRegister.class);
-
-        bind(PublishDiagnosticsReceiver.class).asEagerSingleton();
-        bind(ShowMessageJsonRpcReceiver.class).asEagerSingleton();
```

Comment:
That is a mistake. It also explains why notifications don't work in my current working branch. Good catch!